float64[3] position_error
float64[3] velocity_error